src/audio/buffers/comp_buffer.c

```diff
@@ -152,6 +152,19 @@ static int comp_buffer_sink_set_alignment_constants(struct sof_sink *sink,
 	return 0;
 }
 
+static void comp_buffer_clean(struct sof_audio_buffer *audio_buffer)
+{
+	if (!audio_buffer)
+		return;
+
+	struct comp_buffer *buffer = container_of(audio_buffer, struct comp_buffer, audio_buffer);
+
+	/* reset rw pointers and avail/free bytes counters */
+	audio_stream_reset(&buffer->stream);
```

Comment:
Why would the clean() flows not be part of the reset() API ? We may need to state this too as comments to help with usage.

Reply:
audio_buffer does not have .reset api (yet?)
but @lgirdwood  @lyakh  - I think is a good point - would .reset be better than .clean here?

especially because it's called here:
```
static void buffer_reset_pos(struct comp_buffer *buffer, void *data)
{
	audio_buffer_clean(&buffer->audio_buffer);
}
```

during pipeline prepare operation
```
/* prepare the pipeline for usage */
int pipeline_prepare(struct pipeline *p, struct comp_dev *dev)
{
	struct pipeline_data ppl_data;
	struct pipeline_walk_context walk_ctx = {
		.comp_func = pipeline_comp_prepare,
		.comp_data = &ppl_data,
		.buff_func = buffer_reset_pos,
		.skip_incomplete = true,
	};
```

Reply:
@marcinszkudlinski  Yes - I think keeping a single ```reset()``` keeps the API usage cleaner and simpler to follow. i.e. everyone should understand reset :)